avm/res/network/network-manager/main.bicep

```
@@ -214,58 +233,22 @@ output location string = networkManager.location
 //   Definitions   //
 // =============== //
 
-type lockType = {
-  @sys.description('Optional. Specify the name of lock.')
-  name: string?
-
-  @sys.description('Optional. Specify the type of lock.')
-  kind: ('CanNotDelete' | 'ReadOnly' | 'None')?
-}?
-
-type roleAssignmentType = {
-  @sys.description('Optional. The name (as GUID) of the role assignment. If not provided, a GUID will be generated.')
-  name: string?
-
-  @sys.description('Required. The role to assign. You can provide either the display name of the role definition, the role definition GUID, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
-  roleDefinitionIdOrName: string
-
-  @sys.description('Required. The principal ID of the principal (user/group/identity) to assign the role to.')
-  principalId: string
-
-  @sys.description('Optional. The principal type of the assigned principal ID.')
-  principalType: ('ServicePrincipal' | 'Group' | 'User' | 'ForeignGroup' | 'Device')?
-
-  @sys.description('Optional. The description of the role assignment.')
-  description: string?
-
-  @sys.description('Optional. The conditions on the role assignment. This limits the resources it can be assigned to. e.g.: @Resource[Microsoft.Storage/storageAccounts/blobServices/containers:ContainerName] StringEqualsIgnoreCase "foo_storage_container".')
-  condition: string?
-
-  @sys.description('Optional. Version of the condition.')
-  conditionVersion: '2.0'?
-
-  @sys.description('Optional. The Resource Id of the delegated managed identity resource.')
-  delegatedManagedIdentityResourceId: string?
-}[]?
-
+import { staticMembersType } from 'network-group/main.bicep'
 type networkGroupType = {
   @sys.description('Required. The name of the network group.')
   name: string
 
   @sys.description('Optional. A description of the network group.')
   description: string?
 
-  @sys.description('Optional. Static Members to create for the network group. Contains virtual networks to add to the network group.')
-  staticMembers: {
-    @sys.description('Required. The name of the static member.')
-    name: string
+  @sys.description('Optional. The type of the group member. Subnet member type is used for routing configurations.')
+  memberType: ('Subnet' | 'VirtualNetwork')?
 
-    @sys.description('Required. Resource ID of the virtual network.')
-    resourceId: string
-  }[]?
+  @sys.description('Optional. Static Members to create for the network group. Contains virtual networks to add to the network group.')
+  staticMembers: staticMembersType?
 }[]?
 
-type networkManagerScopeAccessType = ('Connectivity' | 'SecurityAdmin')[]
+type networkManagerScopeAccessType = ('Connectivity' | 'SecurityAdmin' | 'Routing')[]?
```

```suggestion
type networkManagerScopeAccessType = ('Connectivity' | 'SecurityAdmin' | 'Routing')
```
As per soon to be released specs, both the `array` and `nullable` indicators should be set at the parameter, not the UDT-side. The reasoning is simply to make it easier to understand a parameter/property by looking at it.

So this change should go along with updating the parameter to
```bicep
networkManagerScopeAccesses networkManagerScopeAccessType[]?
```

---

Note: The same idea applies to all other types

---

ouch. Let me look into that. Thanks @AlexanderSehr

---

@AlexanderSehr this does require a lot of updates to the module. May I suggest we roll in this new feature please as its a feature request from internal MS eng. Then I can dedicate another PR for this new spec update?

---

Sure thing @ahmadabdalla. Just wanted to mention it as the spec (and test) will come eventually :) Will resolve